mediator/med_phases_prep_glc_mod.F90

```diff
@@ -442,126 +452,96 @@ subroutine med_phases_prep_glc_avg(gcomp, rc)
+         yr_med,mon_med,day_med,sec_med
+    write(logunit,'(a,4(i8,2x))') trim(subname)//'prep glc clock yr, mon, day, sec = ',&
+         yr_prepglc,mon_prepglc,day_prepglc,sec_prepglc
 
     ! If the is ringing - turn it off and continue - otherwise reset field bundle to zero and return
+    call ESMF_ClockGetAlarm(prepglc_clock, alarmname='alarm_glc_avg', alarm=alarm, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
     if (ESMF_AlarmIsRinging(alarm, rc=rc)) then
+       ! Turn off the alarm
        call ESMF_AlarmRingerOff( alarm, rc=rc )
        if (ChkErr(rc,__LINE__,u_FILE_u)) return
-    else
-       call ESMF_LogWrite(trim(subname)//": glc_avg alarm is not ringing - returning", ESMF_LOGMSG_INFO)
-       ! Reset export field bundle to zero
-       call FB_reset(is_local%wrap%FBExp(compglc), value=0.0_r8, rc=rc)
-       if (chkerr(rc,__LINE__,u_FILE_u)) return
-       ! turn on stop timer and return
-       call t_stopf('MED:'//subname)
-       ! return
-       RETURN
-    end if
 
-    !---------------------------------------
-    ! Average import from accumulated land import FB
-    !---------------------------------------
-
-    call ESMF_LogWrite(trim(subname)//": glc_avg alarm is ringing - averaging input from lnd to glc", ESMF_LOGMSG_INFO)
-
-    do n = 1, size(fldnames_fr_lnd)
-       call ESMF_FieldBundleGet(FBlndAccum_lnd, fieldname=fldnames_fr_lnd(n), field=lfield, rc=rc)
-       if (chkerr(rc,__LINE__,u_FILE_u)) return
-       call ESMF_FieldGet(lfield, farrayptr=data2d, rc=rc)
-       if (chkerr(rc,__LINE__,u_FILE_u)) return
-       if (FBlndAccumCnt > 0) then
-          ! If accumulation count is greater than 0, do the averaging
-          data2d(:,:) = data2d(:,:) / real(FBlndAccumCnt)
-       else
-          ! If accumulation count is 0, then simply set the averaged field bundle values from the land
-          ! to the import field bundle values
-          call ESMF_FieldBundleGet(is_local%wrap%FBImp(complnd,complnd), fieldname=fldnames_fr_lnd(n), field=lfield, rc=rc)
+       ! Average import from accumulated land import FB
+       call ESMF_LogWrite(trim(subname)//": glc_avg alarm is ringing - averaging input from lnd to glc", ESMF_LOGMSG_INFO)
+       if (mastertask) then
+          write(logunit,'(a)') trim(subname)//"glc_avg alarm is ringing - averaging input from lnd to glc"
+       end if
+       do n = 1, size(fldnames_fr_lnd)
+          call ESMF_FieldBundleGet(FBlndAccum_lnd, fieldname=fldnames_fr_lnd(n), field=lfield, rc=rc)
           if (chkerr(rc,__LINE__,u_FILE_u)) return
-          call ESMF_FieldGet(lfield, farrayptr=data2d_import, rc=rc)
+          call ESMF_FieldGet(lfield, farrayptr=data2d, rc=rc)
           if (chkerr(rc,__LINE__,u_FILE_u)) return
-          data2d(:,:) = data2d_import(:,:)
+          if (FBlndAccumCnt > 0) then
+             ! If accumulation count is greater than 0, do the averaging
+             data2d(:,:) = data2d(:,:) / real(FBlndAccumCnt)
```

Comment:
should be real(FBlndAccumCnt, kind=r8)

Reply:
Thanks for catching this. Since this might change answers - can we please add it in an upcoming PR?